### DISCUSSION TITLE
SSL: CERTIFICATE_VERIFY_FAILED

### DISCUSSION THREAD

**Discussion by drount** [tiangolo/uwsgi-nginx-flask-docker#367] [Questions] [answered]
Hello, 

after building an image yesterday based on:
`FROM tiangolo/uwsgi-nginx-flask:python3.8` or
`FROM tiangolo/uwsgi-nginx-flask:python3.7`

I got this error when conecting to any HTTPS server:

`ERROR: Unable to download webpage: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1076)> (caused by URLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1076)')))"`

I had to add:
`RUN apt-get update && apt-get install -y ca-certificates`
to my `Dockerfile` in order to fix it. 

Reporting just in case you want to add it to the root image or it helps anybody in the future.

**Comment by sameshl** [comment]
Hey, @drount ! Thanks for giving the fix :blue_heart: . I was facing the exact same problem and this helped a lot.

**Comment by ybizeul** [comment]
Looking at the dates, it seems docker hub should be updated with that fix, but I still have `server certificate verification failed.` when pulling git requirements for example.

**Comment by drount** [comment]
I don't see the changes in the source code

**Comment by ssilverm** [comment]
Had to add this to my own docker file and this seemed to fix it for me:

```
RUN apt-get update
RUN apt-get install -y --reinstall ca-certificates
```

**Comment by gamarobert** [comment]
i believe this is still an issue. above suggestions didnt work for me

**Comment by cleobatista** [comment]
Thank you very much. This helped me a lot.

**Comment by Sana-Rasheed** [comment]
How can I fix this on windows?

**Comment by bignellrp** [comment]
This is still an issue for me too.  Had to run ca-certificates as described to fix

**Comment by tiangolo** [comment, marked as answer]
Thanks for the report! :nerd_face: :cake:
